book-1-orientation/chapters/CLASSES_06-INHERITANCE.md

```diff
@@ -0,0 +1,257 @@
+
+To avoid these kinds of problems, you can to create a more general type in your system and then have all vehicles inherit from it.
+
+## Vehicle Class
+
+Since all of the types are vehicles, a good name for this new type would be `Vehicle`. It contains the properties that were determined to be common to all vehicles in this system. **`Vehicle`** is now the base, or parent, class.
+
+```py
+class Vehicle:
+    def __init__(self):
+        self.main_color = ""
+        self.maximum_occupancy = ""
+```
+
+Then each of the other, more specific, types would inherit from it. The syntax for inheritance is dead simple: Simply specify the parent class in parenthesis after the class definition.
+
+```py
+class Tesla(Vehicle):
+    def __init__(self):
+        self.battery_kwh = 0
+
+    def charge_battery(self):
+        ...
+```
+
+Now any instance of **`Tesla`** will have both of the properties from **`Vehicle`** on it automatically. **`Tesla`** is now a derived, or child, class because it inherits the properties of **`Vehicle`**.
+
+```py
+model_3 = Tesla()
+model_3.main_color = "red"
+
+print(model_3.main_color)
+#prints red
+```
+
+## Lean and Mean Critters
+
+>It's amazing how all that info just came flooding back to you. Dang, those NSS instructors were so great. You really should send them a note sometime to let them know how you're doing. Or buy them lunch at a pricey but tasteful local restaurant. For now, though, it's time to kick some critter code into high gear.
+>
+>All those critter types with identical properties -- that's the first thing to tackle. You decide to consolidate them into a base class. They're all animals, so it makes sense to name it **`Animal`**
+
+```py
+class Animal:
+    def __init__(self, name, species, food, chip_num):
+        self.name = name
+        self.species = species
+        self.food = food
+        self.__chip_number = chip_num
+        self.date_added = date.today()
+
+    def feed(self):
+        print(f'{self.name} was fed {self.food} on {date.today().strftime("%m/%d/%Y")}')
+    
+    @property
+    def chip_number(self):
+        return self.__chip_number
+    
+    @chip_number.setter
+    def chip_number(self, num):
+        pass
+```
+>With all of the properties that were common across all critter types defined in one class, you turn your attention to the existing pile of class definitions. They will need some additions and subtractions to make them all child classes of **`Animal`**. You start with **`Llama`**
+
+**Before:**
+```py
+class Llama:
+
+    def __init__(self, name, species, shift, food):
+        self.name = name
+        self.species = species
+        self.shift = shift
+        self.date_added = date.today()
+        self.walking = True
+        self.food = food
+    
+    def feed(self):
+        print(f'{self.name} was fed {self.food} on {date.today().strftime("%m/%d/%Y")}')
+```
+**After:**
+```py
+# Designate Burro as a child class by adding (Animal) after the class name
```

Comment:
Did you mean LLama?

Reply:
Ah, the downside of commented code. Outdated references.